geoapps/edge_detection/application.py

```diff
@@ -223,148 +223,54 @@ def update_name(self, _):
             self.export_as.value = "Edges"
 
     def compute_trigger(self, _):
-        grid, data = self.get_selected_entities()
-
-        if grid is None or len(data) == 0:
-            return
+        param_dict = self.get_param_dict()
 
-        x = grid.centroids[:, 0].reshape(grid.shape, order="F")
-        y = grid.centroids[:, 1].reshape(grid.shape, order="F")
-        z = grid.centroids[:, 2].reshape(grid.shape, order="F")
-        grid_data = data[0].values.reshape(grid.shape, order="F")
+        param_dict["geoh5"] = self.params.geoh5
+        new_params = EdgeDetectionParams(**param_dict)
+        self.params = new_params
```

Comment:
Same here, let's try to use `self.params.update(param_dict)` instead of generating a new one.